odl/trafos/backends/pywt_bindings.py

```diff
@@ -0,0 +1,532 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Bindings to the ``PyWavelets`` backend for wavelet transforms."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from itertools import product
+import numpy as np
+
+try:
+    import pywt
+    PYWT_AVAILABLE = True
+except ImportError:
+    PYWT_AVAILABLE = False
+
+
+__all__ = ('PAD_MODES_ODL2PYWT', 'PYWT_SUPPORTED_MODES', 'PYWT_AVAILABLE',
+           'pywt_wavelet', 'pywt_mode', 'pywt_coeff_shapes',
+           'pywt_flat_coeff_size',
+           'pywt_flat_array_from_coeffs', 'pywt_coeffs_from_flat_array',
+           'pywt_single_level_decomp', 'pywt_single_level_recon',
+           'pywt_multi_level_decomp', 'pywt_multi_level_recon')
+
+
+PAD_MODES_ODL2PYWT = {'constant': 'zpd',
+                      'periodic': 'ppd',
+                      'symmetric': 'sym',
+                      'order0': 'cpd',
+                      'order1': 'sp1',
+                      'pywt_periodic': 'per'}
+PYWT_SUPPORTED_MODES = PAD_MODES_ODL2PYWT.values()
+
+
+def pywt_wavelet(wavelet):
+    """Convert ``wavelet`` to a ``pywt.Wavelet`` instance."""
+    if isinstance(wavelet, pywt.Wavelet):
+        return wavelet
+    else:
+        return pywt.Wavelet(wavelet)
+
+
+def pywt_mode(pad_mode, pad_const=0):
```

Comment:
Hm, since the function returns the `pywt` padding mode for both styles of modes as input, it would be too strong to call the parameter `odl_pad_mode`. I think I'll stick with just `pad_mode` and change the name of the function to `pywt_pad_mode`.